core/src/main/scala/quasar/destination/s3/impl/DefaultUpload.scala

```diff
@@ -100,7 +102,7 @@ final case class DefaultUpload[F[_]: Concurrent: ContextShift](client: S3AsyncCl
         uploadPartResponse.map(response =>
           CompletedPart.builder.partNumber(partNumber).eTag(response.eTag).build)
       }
-    }).compile.toList
+    }).fold(List.empty[CompletedPart])((acc, partResponse) => acc :+ partResponse)
```

Comment:
Ah, the answer is just one. This is probably reasonable. As an aside, you can replace this with `foldMap(_ :: Nil)`.